File: check_db.js
import { createClient } from '@supabase/supabase-js';
import * as dotenv from 'dotenv';
dotenv.config();

const supabaseUrl = process.env.VITE_SUPABASE_URL || '';
const supabaseKey = process.env.VITE_SUPABASE_ANON_KEY || '';

const supabase = createClient(supabaseUrl, supabaseKey);

async function checkCompanions() {
    console.log("Fetching companions from Supabase...");
    const { data, error } = await supabase.from('companions').select('*');
    if (error) {
        console.error("Error fetching connecting to supabase:", error);
        return;
    }

    console.log(`Total companions in DB: ${data.length}`);
    const missingImages = data.filter(c => !c.image_url || c.image_url.trim() === '');
    console.log(`Companions missing image_url: ${missingImages.length} ->`, missingImages.map(c => c.name));

    const danny = data.find(c => c.name.toLowerCase() === 'danny');
    console.log(`Danny found:`, danny ? true : false, danny ? danny.specialty : '');

    const marias = data.filter(c => c.name.toLowerCase() === 'maria');
    console.log("Maria details: ", marias.map(m => ({ name: m.name, specialty: m.specialty, img: m.image_url })));

    // Try finding the 'Women's Health' specialty
    const womens = data.filter(c => c.specialty.toLowerCase().includes('women'));
    console.log("Women's specialty found: ", womens.map(m => ({ name: m.name, specialty: m.specialty, img: m.image_url })));
}

checkCompanions();

File: check_db.py
import json
import urllib.request
import urllib.parse

URL = "https://qdnctbupmlqhzubwigjn.supabase.co/rest/v1/companions"
KEY = "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJzdXBhYmFzZSIsInJlZiI6InFkbmN0YnVwbWxxaHp1YndpZ2puIiwicm9sZSI6ImFub24iLCJpYXQiOjE3NjE0MTQ0MjEsImV4cCI6MjA3Njk5MDQyMX0.kwfDMX0cGOQ3embdFnQnMNJT27CKi8Krf_Ew8uPgLrU"

req = urllib.request.Request(f"{URL}?select=*", headers={
    "apikey": KEY,
    "Authorization": f"Bearer {KEY}",
    "Range": "0-100"
})

try:
    with urllib.request.urlopen(req) as response:
        data = json.loads(response.read().decode())
        
        print(f"Total companions in live DB: {len(data)}")

        danny = [c for c in data if c['name'].lower() == 'danny']
        print(f"Danny: {json.dumps(danny, indent=2)}")

        womens = [c for c in data if 'women' in c['specialty'].lower()]
        print(f"Women's Health: {json.dumps(womens, indent=2)}")

        missing_img = [c for c in data if not c.get('image_url') or 'http' not in c['image_url']]
        print(f"Missing images count: {len(missing_img)}")
except Exception as e:
    print(e)
